core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -204,35 +209,43 @@ private[spark] class CoarseMesosSchedulerBackend(
 
       for (offer <- offers) {
         val slaveId = offer.getSlaveId.toString
-        val mem = getResource(offer.getResourcesList, "mem")
-        val cpus = getResource(offer.getResourcesList, "cpus").toInt
-        if (totalCoresAcquired < maxCores &&
-            mem >= MemoryUtils.calculateTotalMemory(sc) &&
-            cpus >= 1 &&
+        var totalMem = getResource(offer.getResourcesList, "mem")
+        var totalCpus = getResource(offer.getResourcesList, "cpus").toInt
```

Comment:
not really your code, but we use cpus in some places and cores in others in this file. It would be good to just use "cores" to be consistent.